### PR TITLE
CMR-8732: First cut of the legacy service migration script to support DataQualitySummary migration.

### PR DESCRIPTION
**Pull request by ygliuvt**
In this first cut of the script, it is only coded to work with SIT. We will update the code to work with other environments as we add more data migrations.

### CODE REVIEW

File: legacy-migration/migrate.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+           "MetadataSpecification": {
+               "Name": "DataQualitySummary",
+               "Version": "1.0.0",
+               "URL": "https://cdn.earthdata.nasa.gov/generics/dataqualitysummary/v1.0.0"}}
+
+    header = {"Content-Type": "application/json",
+              "Authorization": access_token,
+              "User-id": "legacy_migration"}
+
+    resp = requests.put(f"{url_root}/ingest/dataqualitysummary/{id}?provider={providers_map[provider_guid]}",
```
Comment (eudoroolivares2016) [inline legacy-migration/migrate.py:43]: With the change of the provider route no longer being a parameter we will need to change the script to
`(f"{url_root}/ingest/providers/providers_map[provider_guid]/dataqualitysummaries/{id}}`

Comment (eudoroolivares2016) [reply legacy-migration/migrate.py:43]: ingest/providers/<Provier-id>/<plural-generic-concept>/<native-id>

**Review by daniel-zamora — APPROVED** [review]

File: legacy-migration/migrate.py (hunk 1 above)
Comment (ygliuvt) [reply legacy-migration/migrate.py]: Cool. updated.

File: legacy-migration/migrate.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+               "URL": "https://cdn.earthdata.nasa.gov/generics/dataqualitysummary/v1.0.0"}}
+
+    header = {"Content-Type": "application/json",
+              "Authorization": access_token,
+              "User-id": "legacy_migration"}
+
+    resp = requests.put(f"{url_root}/ingest/providers/{providers_map[provider_guid]}/dataqualitysummaries/{id}",
+                        data=json.dumps(umm),
+                        headers=header)
+    if resp.status_code > 300:
```
Comment (jceaser) [inline legacy-migration/migrate.py:46]: I saw this and thought >= would be better, I gripped around and did find 301 and no 300 so your probably safe, but if you happen to change for someone else, consider adding the equals.

File: legacy-migration/migrate.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            for r in cursor.execute(sql):
+                providers_map[r[0]] = r[1]
+
+def migrate_row(id, name, summary, provider_guid):
+    global success_count, failure_count
+    umm = {"Id": id,
+           "Name": name,
+           "Summary": summary,
+           "MetadataSpecification": {
+               "Name": "DataQualitySummary",
```
Comment (jceaser) [inline legacy-migration/migrate.py:35]: heads up, this value _may_ change to Data Quality Summary

**Review by jceaser — APPROVED** [review]

**Pull request merged**